mmdet3d/datasets/pipelines/transforms_3d.py

```diff
@@ -905,30 +905,42 @@ class IndoorPatchPointSample(object):
         num_points (int): Number of points to be sampled.
         block_size (float, optional): Size of a block to sample points from.
             Defaults to 1.5.
-        sample_rate (float, optional): Stride used in sliding patch generation.
-            Defaults to 1.0.
         ignore_index (int, optional): Label index that won't be used for the
             segmentation task. This is set in PointSegClassMapping as neg_cls.
             Defaults to None.
         use_normalized_coord (bool, optional): Whether to use normalized xyz as
             additional features. Defaults to False.
         num_try (int, optional): Number of times to try if the patch selected
             is invalid. Defaults to 10.
+        enlarge_size (float | None, optional): Enlarge the sampled patch to
+            [-block_size / 2 - enlarge_size, block_size / 2 + enlarge_size] as
+            an augmentation. If None, set it as 0.01. Defaults to 0.2.
+        min_unique_num (int | None, optional): Minimum number of unique points
+            the sampled patch should contain. If None, use PointNet++'s method
+            to judge uniqueness. Defaults to None.
+
+    Note:
+        This transform should only be used in the training process of point
+            cloud segmentation tasks. For the sliding patch generation and
+            inference process in testing, please refer to the `slide_inference`
+            function of `EncoderDecoder3D` class.
     """
 
     def __init__(self,
                  num_points,
                  block_size=1.5,
-                 sample_rate=1.0,
                  ignore_index=None,
                  use_normalized_coord=False,
-                 num_try=10):
+                 num_try=10,
+                 enlarge_size=0.2,
+                 min_unique_num=None):
         self.num_points = num_points
         self.block_size = block_size
-        self.sample_rate = sample_rate
```

Comment:
We should still accept sample_rate and add a deprecated warning first to inform users to modify their code.

Reply:
Keep sample_rate and add warning.